pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -299,47 +313,12 @@ func (vs *VSphere) getVMByName(ctx context.Context, nodeName k8stypes.NodeName)
 // NodeAddresses is an implementation of Instances.NodeAddresses.
 func (vs *VSphere) NodeAddresses(nodeName k8stypes.NodeName) ([]v1.NodeAddress, error) {
 	// Get local IP addresses if node is local node
-	if vs.localInstanceID == nodeNameToVMName(nodeName) {
+	if vs.hostName == nodeNameToVMName(nodeName) {
 		return getLocalIP()
 	}
-	addrs := []v1.NodeAddress{}
```

Comment:
I observed that each kubelet reports it's own localIP and we actually don't need to find out addresses of some OtherNode.

kubectl describe node node1
shows correct IP addresses.

During testing I even has IPv6 enabled VMs IPv6 addresses were correctly filtered out.